src/consensus/aft/raft.h

```diff
@@ -814,11 +813,17 @@ namespace aft
           (globally_committable ? " committable" : ""),
           hooks->size());
 
+        // Hooks may need to take the state lock if they trigger RPCs (via
+        // frontend::process), so we temporarily release it.
+        state->lock.unlock();
```

Comment:
Releasing the lock here seems dangerous. `frontend::process` would eventually call into Raft again when the underlying transaction is passed back to the consensus for replication. This is something we've encountered before and I believe the best option is to call `frontend::process()` in an async task.